surrealdb/clients/ws.py

```diff
@@ -0,0 +1,284 @@
+"""
+Copyright © SurrealDB Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import asyncio
+from types import TracebackType
+from typing import Any
+from typing import Awaitable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Type
+
+from aiohttp import ClientSession
+from aiohttp import ClientWebSocketResponse
+from aiohttp import WSMsgType
+
+from ..common import json as jsonlib
+from ..common.exceptions import SurrealWebsocketException
+from ..common.id import generate_id
+from ..models import RPCRequest
+from ..models import RPCResponse
+
+__all__ = ("SurrealDBWSClient",)
+
+
+class SurrealDBWSClient:
+    def __init__(
+        self,
+        url: str,
+        *,
+        token: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        namespace: Optional[str] = None,
+        database: Optional[str] = None,
+    ) -> None:
+        self._url = url
+        self._token = token
+        self._username = username
```

Comment:
Why is `_username` private?

Reply:
because they aren't supposed to be publicly accessed, why would they?
it's just used internally for authentication